src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolData.cs

```diff
@@ -2,45 +2,29 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using System.Collections.Immutable;
-using System.ComponentModel;
-using System.Runtime.CompilerServices;
-using Microsoft.VisualStudio.Imaging;
-using Microsoft.VisualStudio.Imaging.Interop;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
 using Microsoft.VisualStudio.Text;
 
 namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
 {
     using SymbolKind = LanguageServer.Protocol.SymbolKind;
 
-    internal sealed class DocumentSymbolData
+    internal sealed record DocumentSymbolData(
+        string Name,
+        SymbolKind SymbolKind,
+        SnapshotSpan RangeSpan,
+        SnapshotSpan SelectionRangeSpan,
+        ImmutableArray<DocumentSymbolData> Children)
     {
-        public string Name { get; }
-        public SymbolKind SymbolKind { get; }
-        public SnapshotSpan RangeSpan { get; }
-        public SnapshotSpan SelectionRangeSpan { get; }
-        public ImmutableArray<DocumentSymbolData> Children { get; }
-
-        public DocumentSymbolData(DocumentSymbol documentSymbol, SnapshotSpan rangeSpan, SnapshotSpan selectionRangeSpan, ImmutableArray<DocumentSymbolData> children)
+        public DocumentSymbolData(
+            DocumentSymbol documentSymbol,
+            SnapshotSpan rangeSpan,
+            SnapshotSpan selectionRangeSpan,
+            ImmutableArray<DocumentSymbolData> children)
+            : this(documentSymbol.Name, documentSymbol.Kind, rangeSpan, selectionRangeSpan, children)
```

Comment:
nit: consider not taking in DocumentSymbol but having the caller break out Name/Kind.  Then this is a pure record without the need for a helper consturctor.